pycyphal/transport/udp/_ip/_v4.py

```diff
@@ -52,40 +54,42 @@ def make_output_socket(
             # Output sockets shall be bound, too, in order to ensure that outgoing packets have the correct
             # source IP address specified. This is particularly important for localhost; an unbound socket
             # there emits all packets from 127.0.0.1 which is certainly not what we need.
-            s.bind((str(self._local), 0))  # Bind to an ephemeral port.
+            s.bind((str(self._local_ip_addr), 0))  # Bind to an ephemeral port.
         except OSError as ex:
             s.close()
             if ex.errno == errno.EADDRNOTAVAIL:
                 raise InvalidMediaConfigurationError(
-                    f"Bad IP configuration: cannot bind output socket to {self._local} [{errno.errorcode[ex.errno]}]"
+                    f"Bad IP configuration: cannot bind output socket to {self._local_ip_addr} [{errno.errorcode[ex.errno]}]"
                 ) from None
             raise  # pragma: no cover
 
         if isinstance(data_specifier, MessageDataSpecifier):
             if remote_node_id is not None:
                 s.close()
-                raise UnsupportedSessionConfigurationError("Unicast message transfers are not defined.")
+                raise UnsupportedSessionConfigurationError("Message transfers don't require a remote_node_id.")
```

Comment:
I think this should be promoted to an assertion check because an attempt to instantiate a unicast subject port indicates that something is broken in the higher layers.